api/resources_portal/management/commands/material_additional_metadata_key_change.py

```diff
@@ -0,0 +1,183 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.utils import timezone
+
+from resources_portal.config.logging import get_and_configure_logger
+from resources_portal.models import Material
+
+logger = get_and_configure_logger(__name__)
+
+
+def validate_category(category):
+    return len([choice for choice in Material.CATEGORY_CHOICES if category in choice]) > 0
+
+
+def paginate_queryset(query_set, function, pagination):
+    current_page = 0
+    size = len(query_set)
+    current_slice = []
+    materials_updated = 0
+
+    while current_page < size:
+        if (size - current_page) < pagination:
+            current_slice = query_set[current_page : current_page + (size - current_page)]
+            current_page += size - current_page
+        else:
+            current_slice = query_set[current_page : current_page + pagination]
+            current_page += pagination
+
+        current_materials = []
+        for material in current_slice:
+            current_material = function(material)
+            current_material.updated_at = timezone.now()
+            current_materials.append(current_material)
+
+        Material.objects.bulk_update(current_materials, ["additional_metadata", "updated_at"])
+        materials_updated += len(current_materials)
+        logger.info(
+            f"{len(current_materials)} materials processed. {materials_updated} total processed so far."
+        )
+
+    return materials_updated
+
+
+def log_nonapplicable_materials(query_set, key):
+    if len(query_set) > 0:
+        logger.warning(
+            f"The key '{key}' was not present in {len(query_set)} materials. The IDs of these materials are: "
+        )
+
+        idString = ""
+        for material in query_set:
+            idString += str(material.id) + ", "
+
+        idString = idString.strip(", ")
+        logger.warning(idString)
+
+
+def add_key(category, new_key, default, pagination):
+    query_set = None
+    materials_changed = 0
+
+    if category:
+        query_set = Material.objects.filter(category=category)
+    else:
+        query_set = Material.objects.all()
+
+    def add_func(material):
+        material.additional_metadata[new_key] = default
+        return material
+
+    materials_changed = paginate_queryset(query_set, add_func, pagination)
+
+    logger.info(
+        f"The key '{new_key}' was added to {materials_changed} materials. The default value was '{default}'."
+    )
+
+
+def remove_key(category, old_key, pagination):
+    query_set = None
+    materials_changed = 0
+
+    if category:
+        query_set = Material.objects.filter(category=category)
+    else:
+        query_set = Material.objects.all()
+
+    filtered_query_set = query_set.filter(additional_metadata__has_key=old_key)
+    not_applicable_query_set = query_set.exclude(additional_metadata__has_key=old_key)
+
+    # Evaluates the queryset. Querysets are usually lazy, this forces evaluation
+    len(not_applicable_query_set)
```

Comment:
Why do we need the queryset evaluated?

Reply:
When using .count(), we don't. I refactored to remove this.